scripts/extensions/sams/src/components/samsWorkspace.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+
+import {ISuperdesk} from 'superdesk-api';
+import {ISamsAPI, MODAL_TYPES} from '../interfaces';
+
+import {ButtonGroup, Dropdown, SubNav} from 'superdesk-ui-framework/react';
+import {IMenuGroup} from 'superdesk-ui-framework/react/components/Dropdown';
+import {HeaderPanel, LayoutContainer, MainPanel} from '../ui';
+
+import {getManageSetsModalComponent} from './sets/manageSetsModal';
+
+interface IState {
+    currentModal: MODAL_TYPES;
```

Comment:
There will be more modals added to this workspace in the future

Reply:
In the case of more than one modal it might not be possible to keep the function stateless, but still, it could be a simple function call without a need to keep selected modal in the component state. Think notifications, for example, you call the function and a notification gets displayed. No need to store in the component state which one is being displayed at the moment.